### ISSUE TITLE
`Style/MultilineTernaryOperator` Error

### ISSUE THREAD

**Issue by a-lavis** [rubocop/rubocop#12691]
## Expected behavior

When running `rubocop -a foo.rb` where `foo.rb` is the following file:

```ruby
# frozen_string_literal: true

Foo = Data.define(:input) do
  def one?
    input == 1
  end

  def two?
    input == 2
  end

  def three?
    input == 3
  end
end

def calculate(foo_input)
  foo = Foo.new(input: foo_input)

  foo.one? ? 'one!' :
    foo.two? ? 'two!' :
      foo.three? ? 'three!' : 'other number...'
end

puts calculate(9)
```
I expect RuboCop to not encounter an error.

## Actual behavior

When running `rubocop -a foo.rb`, I see the following output:
```
Inspecting 1 file
An error occurred while Style/MultilineTernaryOperator cop was inspecting /Users/{username}/foo.rb:21:4.
To see the complete backtrace run rubocop -d.
W

Offenses:

foo.rb:20:3: C: [Corrected] Style/MultilineTernaryOperator: Avoid multi-line ternary operators, use if or unless instead.
  foo.one? ? 'one!' : ...
  ^^^^^^^^^^^^^^^^^^^
foo.rb:21:3: C: [Corrected] Layout/IndentationWidth: Use 2 (not 0) spaces for indentation.
  'one!'

foo.rb:21:5: C: [Corrected] Style/NestedTernaryOperator: Ternary operators must not be nested. Prefer if or else constructs instead.
    foo.two? ? 'two!' : ...
    ^^^^^^^^^^^^^^^^^^^
foo.rb:22:1: C: [Corrected] Layout/ElseAlignment: Align else with if.
else
^^^^
foo.rb:23:3: C: [Corrected] Layout/IndentationWidth: Use 2 (not 0) spaces for indentation.
  if foo.two?

foo.rb:23:3: C: [Corrected] Style/IfInsideElse: Convert if nested inside else to elsif.
  if foo.two?
  ^^
foo.rb:23:3: C: [Corrected] Style/MultilineTernaryOperator: Avoid multi-line ternary operators, use if or unless instead.
  foo.two? ? 'two!' : ...
  ^^^^^^^^^^^^^^^^^^^
foo.rb:24:7: C: [Corrected] Layout/ElseAlignment: Align else with if.
      else
      ^^^^
foo.rb:24:7: C: [Corrected] Style/NestedTernaryOperator: Ternary operators must not be nested. Prefer if or else constructs instead.
      foo.three? ? 'three!' : 'other number...'
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
foo.rb:25:1: C: [Corrected] Layout/ElseAlignment: Align else with if.
else
^^^^
foo.rb:25:1: W: [Corrected] Layout/EndAlignment: end at 25, 0 is not aligned with if at 20, 2.
end
^^^
foo.rb:25:3: C: [Corrected] Layout/IndentationWidth: Use 2 (not 6) spaces for indentation.
        foo.three? ? 'three!' : 'other number...'
  ^^^^^^
foo.rb:25:5: C: [Corrected] Layout/IndentationWidth: Use 2 (not -2) spaces for indentation.
    foo.three? ? 'three!' : 'other number...'
    ^^
foo.rb:26:7: W: [Corrected] Layout/EndAlignment: end at 26, 6 is not aligned with if at 20, 2.
      end
      ^^^
foo.rb:27:1: W: [Corrected] Layout/EndAlignment: end at 27, 0 is not aligned with if at 23, 2.
end
^^^

1 file inspected, 15 offenses detected, 15 offenses corrected

1 error occurred:
An error occurred while Style/MultilineTernaryOperator cop was inspecting /Users/{username}/foo.rb:21:4.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.60.2 (using Parser 3.3.0.5, rubocop-ast 1.30.0, running on ruby 3.3.0) [arm64-darwin23]
```

## Steps to reproduce the problem

Run `rubocop -a foo.rb` where `foo.rb` is the above file.

## RuboCop version

```
> rubocop -V
1.60.2 (using Parser 3.3.0.5, rubocop-ast 1.30.0, running on ruby 3.3.0) [arm64-darwin23]
```
